web/gui/main.js

```diff
@@ -2832,39 +2834,95 @@ function versionsMatch(v1, v2) {
     if (v1 == v2) {
         return true;
     } else {
-        var s1=v1.split('-');
-        var s2=v2.split('-');
-        if (s1.length !== s2.length) return false;
-        if (s1.length === 4) s1.pop();
-        if (s2.length === 4) s2.pop();
-        return (s1.join('-') === s2.join('-'));
+        let s1=v1.split('.');
+        let s2=v2.split('.');
+        // Check major version
+        let n1 = parseInt(s1[0].substring(1,2),10);
+        let n2 = parseInt(s2[0].substring(1,2), 10);
+        if ( n1 < n2 ) return false;
+        else if ( n1 > n2 ) return true;
+
+        // Check minor version
+        n1 = parseInt(s1[1],10);
+        n2 = parseInt(s2[1],10);
+        if ( n1 < n2 ) return false;
+        else if ( n1 > n2 ) return true;
+
+        // Split patch: format could be e.g. 0-22-nightly
+        s1=s1[2].split('-');
+        s2=s2[2].split('-');
+
+        n1 = parseInt(s1[0],10);
+        n2 = parseInt(s2[0],10);
+        if ( n1 < n2 ) return false;
+        else if ( n1 > n2 ) return true;
+
+        n1 = (s1.length > 1)?parseInt(s1[1],10):0;
```

Comment:
Minor formatting note, pls add some spacing around `?` and `:`.

Reply:
Fixed it